core/math/eigen_value_symmetric.h

```diff
@@ -0,0 +1,145 @@
+/*  eigen_value_symmetric.h                                               */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#include "core/math/basis.h"
+#include "core/math/vector3.h"
+
+// This code is simply a re-implementation of the same function from Jolt Physics,
+// but using Godot's Vector3 and Basis structures
+bool eigen_value_symmetric(const Basis &in_matrix, Basis &out_eig_vec, Vector3 &out_eig_val) {
+	const int MAX_SWEEPS = 50;
+	const int N = 3;
+
+	Basis a = in_matrix;
+
+	Vector3 b, z;
+
+	for (int ip = 0; ip < N; ++ip) {
+		// Initialize b and output to diagonal of a
+		b[ip] = a[ip][ip];
+		out_eig_val[ip] = a[ip][ip];
+
+		// reset z
+		z[ip] = 0.0;
+	}
+
+	for (int sweep = 0; sweep < MAX_SWEEPS; ++sweep) {
+		// sum the off-diagonal elements of a
+		real_t sm = 0.0;
+		for (int ip = 0; ip < N; ++ip) {
+			for (int iq = ip + 1; iq < N; ++iq) {
+				sm += abs(a[ip][iq]);
```

Comment:
```suggestion
				sm += Math::abs(a[ip][iq]);
```
For all, please use the built-in methods